### PR TITLE
Enable runsc compatLog support on arm64.

### PR DESCRIPTION
**Pull request by xiaobo55x**
Signed-off-by: Haibo Xu <haibo.xu@arm.com>
Change-Id: I3fd5e552f5f03b5144ed52647f75af3b8253b1d6

### CODE REVIEW

**Comment by xiaobo55x** [comment]
Ref #63.

**Comment by xiaobo55x** [comment]
Hi @fvoznika,

Please help review this patch!

Thanks,
Haibo

File: runsc/boot/compat.go [hunk 1]
```diff
@@ -86,36 +84,8 @@ func (c *compatEmitter) Emit(msg proto.Message) (bool, error) {
 }
 
 func (c *compatEmitter) emitUnimplementedSyscall(us *spb.UnimplementedSyscall) {
-	regs := us.Registers.GetArch().(*rpb.Registers_Amd64).Amd64
-
-	c.mu.Lock()
-	defer c.mu.Unlock()
-
-	sysnr := regs.OrigRax
```
Comment (fvoznika) [inline runsc/boot/compat.go]: You could keep this method here and create a function to return the syscall number from regs in the arch specific files.

File: runsc/boot/compat_test.go [hunk 2]
```diff
@@ -32,54 +30,3 @@ func TestOnceTracker(t *testing.T) {
 		}
 	}
 }
-
-func TestArgsTracker(t *testing.T) {
```
Comment (fvoznika) [inline runsc/boot/compat_test.go]: You could keep the bulk of the test in this file. Replace rdi/rsi with arg1/arg2 and create a function to convert args into registers in the arch specific file.

File: runsc/boot/compat_test.go [hunk 3]
```diff
@@ -32,54 +30,3 @@ func TestOnceTracker(t *testing.T) {
...
-
-			regs.Rdi, regs.Rsi = tc.rdi2, tc.rsi2
-			if got := c.shouldReport(regs); tc.want != got {
-				t.Errorf("second call to shouldReport, got: %t, want: %t", got, tc.want)
-			}
-		})
-	}
-}
-
-func TestArgsTrackerLimit(t *testing.T) {
```
Comment (fvoznika) [inline runsc/boot/compat_test.go]: Same here

File: runsc/boot/compat.go (hunk 1 above)
Comment (xiaobo55x) [reply runsc/boot/compat.go]: The "syscall number" is not the only difference here, and the syscall.SYS_ARCH_PRCTL is not supported on arm64.

File: runsc/boot/compat_test.go (hunk 2 above)
Comment (xiaobo55x) [reply runsc/boot/compat_test.go]: Good idea! I will replace them in the next patch.

File: runsc/boot/compat.go (hunk 1 above)
Comment (fvoznika) [reply runsc/boot/compat.go]: Yes, making a call to an arch specific function if there are no matches here would be preferable to reduce duplication as much as possible.

Comment (xiaobo55x) [reply runsc/boot/compat.go]: Done, thanks!

File: runsc/boot/compat_test.go (hunk 3 above)
Comment (xiaobo55x) [reply runsc/boot/compat_test.go]: Done!

**Pull request merged**